crates/resolver/src/caching_client.rs

```diff
@@ -286,31 +286,19 @@ where
         response_code: ResponseCode,
         trusted: bool,
     ) -> ProtoError {
-        if valid_nsec || !is_dnssec {
-            // only trust if there were validated NSEC records
-            ProtoErrorKind::NoRecordsFound {
-                query: Box::new(query),
-                soa: soa.map(Box::new),
-                ns,
-                negative_ttl,
-                response_code,
-                trusted: true,
-                authorities: None,
-            }
-            .into()
-        } else {
-            // not cacheable, no ttl...
-            ProtoErrorKind::NoRecordsFound {
-                query: Box::new(query),
-                soa: soa.map(Box::new),
-                ns,
-                negative_ttl: None,
-                response_code,
-                trusted,
-                authorities: None,
-            }
-            .into()
+        ProtoErrorKind::NoRecordsFound {
+            query: Box::new(query),
+            soa: soa.map(Box::new),
+            ns,
+            negative_ttl,
```

Comment:
The previous logic unconditionally set `negative_ttl` to `None` for the other branch, where now it gets populated with the arg that might be `Some(_)`. Was that intentional or should it also consider `valid_nsec` & `is_dnssec`?

Reply:
Ohh, good catch. That's probably intentional.